src/modules/sdlog2/sdlog2.c

```diff
@@ -584,6 +596,12 @@ static void *logwriter_thread(void *arg)
 		if (++poll_count == 10) {
 			fsync(log_fd);
 			poll_count = 0;
+
+			/* check if space is available, if not stop everything */
+			if (check_free_space() != OK) {
```

Comment:
And I don't really understand what happens in statfs either. What about slowing it down even more to be sure?